### ISSUE TITLE
The entry point for locally served files is incorrectly presumed when consumed on a nonlocal server

### ISSUE THREAD

**Issue by Adventive** [ampproject/amphtml#12747]
## What's the issue?
The entry point for locally served files is incorrectly presumed (in `extension-location.js`) via the location object when consuming them on a nonlocal server. 

## How do we reproduce the issue?
- gulp
  - no --host config
- run secondary server 
- consume amp extension
  - Note: issue may only be visible with extensions that use web workers
  - In my scenario, I used `amp-bind`. I followed from the `AmpWorker` constructor to  `calculateEntryPointScriptUrl()`. I noticed it generated `ww.js` instead of `ww.max.js` with `serve=default` and that it was trying to load from my second server instead of localhost.

## What browsers are affected?
- all browsers

## Which AMP version is affected?
- version: 1516393161691

------
#### Potential fix
Add in check (probably in `calculateScriptBaseUrl()`) for whether or not a host config has been changed. If it has, proceed as it is currently, otherwise use `localhost`. Probably also check if port config has been changed.

**Comment by rsimha-amp** [comment]
I believe the problem is in `gulpfile.js`. When `gulp build` is run, we call `buildWebWorker()`, which builds `ww.max.js`, while when `gulp` or `gulp watch` are run, we don't make that call. Fix coming up.

**Comment by rsimha-amp** [comment]
@Adventive this should be fixed by #12900. I tested to make sure that `amp-bind` works after running `gulp` on a clean client. Let me know if this is still broken for you.

**Comment by Adventive** [comment]
@rsimha-amp 
I pulled in the latest changes and performed `gulp clean` before running `gulp` again but unfortunately the issue is still there.
Where and what version of the file it is trying to load is the issue.

<img width="1299" alt="runtime" src="https://user-images.githubusercontent.com/34548005/35169126-568ddedc-fd29-11e7-8813-2544de5fd231.png">
<img width="823" alt="scripts" src="https://user-images.githubusercontent.com/34548005/35169127-569e7666-fd29-11e7-817f-df2ea8e2e5c4.png">
<img width="1440" alt="errors" src="https://user-images.githubusercontent.com/34548005/35169185-911ee974-fd29-11e7-86b9-75c3c6c9bf1d.png">
<img width="282" alt="generatedfiles" src="https://user-images.githubusercontent.com/34548005/35169354-3ffe3c9c-fd2a-11e7-819b-3c0a90ddbaa3.png">

**Comment by Adventive** [comment]
I believe the solution may be similar to that of what is implemented in `3p-frame.js : getAdsLocalhost()` except, including the fix that for #12741.
<img width="1432" alt="emptyports" src="https://user-images.githubusercontent.com/34548005/35169722-9c368734-fd2b-11e7-9389-44e1014e2c63.png">

**Comment by rsimha-amp** [comment]
@Adventive Thanks for the heads up. Taking a look at #12741 now.

**Comment by rsimha-amp** [comment]
@Adventive Can you give me the exact repro steps? I'm not sure how you started a secondary server, or why you had to do so.

**Comment by Adventive** [comment]
@rsimha-amp The secondary server is our own development server. So you really just need any secondary server to consume the `localhost` served files.

**Comment by Adventive** [comment]
Can this get re-opened since it was not fixed by #12900

**Comment by rsimha** [comment]
@westonruter Do you know if your PR #24066 happens to have fixed this old issue?

**Comment by westonruter** [comment]
@rsimha I'm not sure. The `http://ads.localhost` URL seems like it would cause problems when running `gulp server` with `--https` since it would cause mixed content security warning. Should not the `ads.localhost` be configurable (such as in `gulp serve`) to point to an actual host? Is this something that is missing from the [E2E getting started instructions](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server)?

**Comment by westonruter** [comment]
@rsimha I think I'm seeing the same thing here. I've got the latest from `master` pulled down. I'm running the generic `gulp --https` on my machine, and I've got the Local AMP Proxy extension running with `https://localhost:8000` applied. When attempting to access my AMP-first site at https://weston.ruter.net/ I am seeing in the console:

![image](https://user-images.githubusercontent.com/134745/66250090-a23ea780-e6f2-11e9-820a-b1126fca5fa7.png)

For some reason, `calculateExtensionScriptUrl()` is returning a URL that is using the current page as the base, as opposed to the AMP CDN base URL.
